sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -144,8 +144,11 @@ object SpillableColumnarBatch extends Arm {
         (0 until numColumns)
             .forall(i => batch.column(i).isInstanceOf[GpuColumnVectorFromBuffer])) {
       val cv = batch.column(0).asInstanceOf[GpuColumnVectorFromBuffer]
-      withResource(GpuColumnVector.from(batch)) { table =>
-        RapidsBufferCatalog.addTable(id, table, cv.getBuffer, initialSpillPriority)
+      withResource(batch) { batch =>
```

Comment:
While working with compressed batches, I discovered there's another problem with SpillableColumnarBatch.  It does not close the batch if it's a compressed column batch.  Essentially we need to close the batch regardless of what type of batch it is.  And similar to this case, we need to make sure the underlying device buffer refcount gets incremented accordingly.